test/extended/util/framework.go

```diff
@@ -1691,15 +1691,16 @@ type podExecutor struct {
 
 // NewPodExecutor returns an executor capable of running commands in a Pod.
 func NewPodExecutor(oc *CLI, name, image string) (*podExecutor, error) {
-	out, err := oc.Run("run").Args(name, "--labels", "name="+name, "--image", image, "--restart", "Never", "--command", "--", "/bin/bash", "-c", "sleep infinity").Output()
-	if err != nil {
-		return nil, fmt.Errorf("error: %v\n(%s)", err, out)
-	}
-	_, err = WaitForPods(oc.KubeClient().CoreV1().Pods(oc.Namespace()), ParseLabelsOrDie("name="+name), CheckPodIsReady, 1, 3*time.Minute)
-	if err != nil {
-		return nil, err
-	}
-	return &podExecutor{client: oc, podName: name}, nil
+	execPod := e2epod.CreateExecPodOrFail(oc.AdminKubeClient(), oc.Namespace(), name, func(pod *corev1.Pod) {
```

Comment:
Why not use the OpenShift-specific `CreateExecPodOrFail` that already configures the name, image, command, and security context? https://github.com/openshift/origin/blob/f825ad788c95df1b805c89cbdfe912559bf575c2/test/extended/util/pods.go#L73-L91

(By the way, is that comment "Pod name is uniquely generated." true when when the tweak function clears `pod.GenerateName` and overrides `pod.Name`?)

Going a step further, you could set `TerminationMessagePolicy` in the OpenShift-specific `CreateExecPodOrFail` for the benefit of all its callers.  However, that would greatly increase the blast radius of this PR; the safer option would be to use the OpenShift-specific `CreateExecPodOrFail` function and layer on a tweak function that sets `TerminationMessagePolicy` just for this test.

Reply:
> Why not use the OpenShift-specific CreateExecPodOrFail that already configures the name, image, command, and security context?

I did originally, but David mentioned to use the upstream method:
https://github.com/openshift/origin/pull/27635#discussion_r1064728173

> Going a step further, you could set TerminationMessagePolicy in the OpenShift-specific CreateExecPodOrFail for the benefit of all its callers. 

Right, in a previous revision of the PR, when I used the openshift-specific `CreateExecPodOrFail` and set it for all E2E tests.

I think we need @deads2k to help us understand what the advantage of using the upstream `CreateExecPodOrFail` vs. openshift-specific `CreateExecPodOrFail`. Thoughts?

Reply:
Just for reference, David got back to me on his motivation on using upstream functions, and not our wrappers:

> I would choose consistency with upstream so that pod execution follows a consistent path. Upstream has been better than us about keeping the helpers functioning well and making them more intelligent.  I'd like to strip back and remove the openshift helpers.